### PR TITLE
Configurator GUI rework phase 1

### PR DESCRIPTION
**Pull request by skaman82**
Don't merge now. Still have to do lot of work in the next weeks. Just announcing. Some help with JS would be very appreciated.
bildschirmfoto 2015-10-29 um 20 46 44

### CODE REVIEW

**Comment by hydra** [comment]
please rebase against current development branch, not master branch. looking good so far

**Comment by hydra** [comment]
just for fun i tried merging this against master and noted that the html and formatting in your commits makes it very hard to follow the format of the html.

whever possible a closing element should be:
a) on the same line of code as the opening element
or 
b) at the same tab index as the opening element.

Good:
```
<div><p>text</p></div>

<div>
  <p>text</p>
</div>

<div>
  <p>
    text
  </p>
</div>
```

Bad:
```
<div><p>text
              </p>
                </div>


<div>
<p>text</p></div>


<div><p>text</p>
</div>

```

This is CRITICAL otherwise is makes merging the changes very very hard and difficult to follow.

**Comment by hydra** [comment]
@skaman82 Don't work on the LED Strip tab code yet.  Speak to Destos on IRC as he is doing some work on it.  We don't want to end up with duplicate or rework.

**Comment by hydra** [comment]
The cleanflight branding is very important to the project.  Some tabs were using watermarks.  
watermark goals:
1. the watermark should not get in the way.  
2. screenshots (even resized smaller ones) must have the instantly recognisable logo.

https://www.google.co.uk/search?q=cleanflight&source=lnms&tbm=isch&sa=X&ved=0CAgQ_AUoAmoVChMI5Lmgi6vtyAIVyT8aCh0YFwX5&biw=2560&bih=1303

e.g. It's should be easy to see which are the cleanflight screenshots in pages like that.

**Comment by hydra** [comment]
There is a script that generates the motor mixer images.  This must be updated to use the arrows and color scheme and then simple rebuild all images for all models.

**Comment by hydra** [comment]
Also, be aware that there are lots of how-to videos on youtube so users should be able to navigate the UI without too much effort.  e.g. don't move UI elemements from one tab to another (yet).

**Comment by skaman82** [comment]
@hydra 

Thank you for having a look at this. My reply in short:

1. Rebasing: Don't know how to do that. Using Github for the firs time. Do I need to make an new pull request for that?
2. HTML-Stucture: Yes I will clean up the HTML structure when I'm ready - this is just now work in progress
2. LED-Tab: OK, will leave the LED Tab alone for now
3. Watermarks: If someone on my team at work would place a watermark in a application window I would slap him in the neck for doing that ;-) Watermarks look always very silly and are mostly used by people without graphical education. Do we really need them? We would have a clearly visible big logo in the header instead
4. Motor order images: Where is the script located that generates the motor order SVGs? I could do them all manually in Illustrator.
5. Don't move elements from one tab to another: OK, noted.

**Comment by stronnag** [comment]
At hydra's request, I tested this on some 'disadvantaged' platforms (the sort one might take to the field for tuning). I branched the current dev build from e2df936 and the pulled this PR from https://github.com/skaman82/cleanflight-configurator.git to do this.

My notes are ~~attached~~ linked  [PDF analysis](http://www.zen35309.zen.co.uk/cf-ui/jhnotes-cfui.pdf)   (as one cannot, alas, add attachments); at the time this seemed the easiest way to provide a comparison of the extant and rewrite UIs.

Exec summary: Less usable on small screens, some buggy UI artefacts on ARM.

**Comment by hydra** [comment]
@skaman82 no problem.

1. As there have been other changes to master since you started working on this you will have to correctly merge the changes into your new code.

basically your PR targets the `master` branch instead of the `development`.  You will need to checkout the current development branch and merge all your changes into it (fixing all the problems at the end).
 
OR rebase from development and merge your changes as each rebased commit is applied.

If you've never used git before then this might be a bit head-scratching and you might find the merge easier than the rebase.

If it's easier for you to make a new PR that's fine, but generally you would just do a force push onto your PR branch (GUI-rework) on your github repo from yor machine.  If in doubt backup your cleanflight-configurator folder include the .git folder first, then you can always restore it if you get messed up.

I could do it but since the code formatting is sub-optimal, as I noted above, I gave after after a few minuites because it would have required more time than I had.  You're the best person to do it since you know about your changes.

2. please clean it up NOW on your branch if you can, do it BEFORE you attempt to merge  It will be so much easier if you do.

4. might be a personal preference thing.  I love the watermark, especially on the CLI tab.  Not had any negative feedback regarding it.  Do not underestimate the power of a brand.  Giant logo helps to re-inforce it in people's brains!

5. `cleanflight-configurator/support/svg_model_motors.rb`

It's a ruby script that @stronnag created.  If you need help I'm sure stronnag will be happy to assist.  He might even be able to make the changes for you and generate new images if you need him to and he's not to busy.



7. Please take a look at the feedback from @stronnag and attend it to the issues he raises.  Guessing the amount of padding in the header and in the main tab area is likely the cause of the lack of vertical space.  Pay attention to the maximum window size stronnag can use and design for that as a minimum.  Avoid sideways scrolling areas.

8. I prefer to be able to see the log file all the time.   Perhaps it should be OPEN by default.  I do like the expanding/contracting log, it's quite nice.  I think we need to check to see if there's anything important that users might miss before decided it should be closed by default.  Certainly we can make it remember the opened/closed setting on a per-user basis at a later date.

**Comment by hydra** [comment]
9. feel free to change the colors of anything if that helps create  a consistent feel to the GUI.  e.g. colors of receiver bars.  colors of graphs.  colors of alternating lines and indicators on the adjustments tab.  colors of table header backgrounds.  colors of motor/servo bar graphs on the motors tab.  

10. the baro icon looks like a temperature guage rather than a barometer/height/pressure sensor indicator.

**Comment by skaman82** [comment]
@stronnag Thank you for testing this. The vertical space for the tab content is almost the same as it is in the current UI. To have a more visual appealing style and cleaner look I had to add more space between some elements. This is a balance between "nice" and "as much content as you can get in". I can reduce some spacing here and there but to design this I simply need little more space than it had before but I'll try my best to meet the requirements for small screens.

Thank you for noticing that on some systems there are only 3 digit values visible in the config tab. Thats clearly a bug and I will fix that.

Could you also help me by modifying the svg_model_motors.rb script to have the models look more like my quadX model generated in Illustrator?

**Comment by skaman82** [comment]
@hydra 

1. Rebasing: Still have not a clue how to do this. When I look it up on google how to do this I get some instructions how to do it with terminal. Never used terminal before, looks very scary :)
2. HTML Cleanup: Done! Reworked the format of all HTML files I've touched so far.
7. Feedback from @stronnag on small screen devices: I think I have a good idea how to make this better. Have to test this first.
8. Log window: It would be great if somebody can add a script that remembers the log window state (opened/cosed).
9: Colors change: Will do. Wanted to do this anyway :)
10. Baro icon: Yeah I know. Do you have a better idea how it could like? I can make a new icon if I have a reference.

**Comment by skaman82** [comment]
@hydra 
4: If you like it so much I will readd the logo watermark in the cli window :)

**Comment by skaman82** [comment]
@stronnag I have modified some styles and the content now adapts depending on current window size. Could you test this again please? I've modified just the SETUP TAB completely. Other tabs need further modification if that works good. Could you also please give me your feedback and the max resolution of the devices you test with? Do not no yet where to set the breaking point to the media-query.

**Comment by stronnag** [comment]
Unfortunately, my git foo is about equal to http://www.xkcd.com/1597/, so I have absolutely zero clue as to how I might apply your latest work to the branch I created from https://github.com/cleanflight/cleanflight-configurator/commit/e2df936491ecde0be4118225c8f020aea33f99c8 and your previous update, without blowing the whole thing away and starting again (which I guess is what I will have to do ...)

**Comment by skaman82** [comment]
then we are on the same page about github :)

**Comment by tracernz** [comment]
Try SmartGit. ;-)

**Comment by stronnag** [comment]
````$ pacman -Ss smartgit
````
No results. Or was the comment self-referential?

**Comment by tracernz** [comment]
It's a GUI app, free for non-commercial use. The only time I go to the terminal for Git is to edit or squash commits. http://www.syntevo.com/smartgit/

**Comment by skaman82** [comment]
@prodrone @stronnag 
Experiment: Integrated a collapsed view of the navigation bar vor small displays. Just resize the window to its minimum size. Feedback welcome!

bildschirmfoto 2015-11-01 um 22 35 34

**Comment by ProDrone** [comment]
Looking good! But i am missing the texts when only icons are shown. Is it possible when you hover over them to display the tab text in a popup balloon?

Rebasing against developer branch is still a thing that will give you some headache... With every change/commit you make it will become harder. In my view you should do that first before you continue. Unfortunately i cannot help with that because i am not all that familiar with the configurator code and especially have no knowledge about the changes you made.

About a week ago i tried this (to see if i could make it right) but i got completely lost...

The tools that i use are:
SourceTree from Atlassian : https://www.atlassian.com/software/sourcetree/overview
and
Perforce P4Merge for merging : https://danlimerick.wordpress.com/2011/06/19/git-for-window-tip-use-p4merge-as-mergetool/

**Comment by skaman82** [comment]
@hydra I think I am all over my head with this rebasing. This is my only project on Github and I have no problem to give my login info to somebody who can actually do this rebase thing. Just write me a short E-Mail to albert@artdivision.de.

**Comment by wanderzell** [comment]
@skaman82 I can try to help you with rebasing if you just give me bit more time (it's monday :) Btw. the responsive approach to screen size issue is :+1: I will test it through and improve if neccessary. I also think we should do same with the header, at some point. For example - if there is connection we really don't need to see the host/port anymore. Likewise if there is no connection, there is little point of showing the "sensors". That way we also get valuable screen real estate to the header making it more aware about the current state of the UI.

But lets keep all that new, fancy stuff in separate future discussion. I think the most important now is to keep our work focused and scope clearly defined. Hence we can define the first PR as **Layout and skin update** that will contain new screen layout and visual styles, but do NOT include any behavioral changes. We can include the improvements piece by piece in later PR-s.

**Comment by skaman82** [comment]
@wanderzell Making the header elements behaving accordingly to the connection state is pretty easy. I've done this in 10 minutes with four lines of code.

**Comment by wanderzell** [comment]
@skaman82 sure, it's easy. Just there was no apparent reason just yet as everything fit well. My comment was mostly regarding the plans to include the Rateprofile selection and Dataflash elements to the header. Both ideas are very welcome.

I checked current responsive behaviour in the latest code. While I like the menu collapse as nice way how to maximize the usable screen area, I don't think it is good idea to make the text smaller on smaller screens. We make our life harder: 1) text not that well readable 2) all new screens then need to be tested and possibly adjusted with the different font sizes. But I hardly see any benefits.

The content area width on 1024px screen is of same as it is on 1200px because sidebar is collapsed. I'm sure all layouts could (and should) be adjusted to make sure this minimum content area width does not require any horizontal scrolling.

**Comment by skaman82** [comment]
@wanderzell You wrote: "I'm sure all layouts could (and should) be adjusted to make sure this minimum content area width does not require any horizontal scrolling." Do you see any content that require horizontal scrolling? So far I as can see everything fits.

Regarding rebasing: the state of my files should be equivalent to the development branch of the configurator before this was changed:
bildschirmfoto 2015-11-02 um 15 53 10

**Comment by skaman82** [comment]
Opened a new pull request with correct target branch. To everybody who is following: please move on to Configurator GUI rework phase 1 #266
